test/functional/transactions/0.tx.js

```diff
@@ -0,0 +1,1089 @@
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Value -1 is less than minimum 0');
+					done();
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('using float should fail', function (done) {
+				transaction.fee = 1.2;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Expected type integer but found type number');
+					badTransactions.push(transaction);
+					done();
+				});
+			});
+
+			it('using negative float should fail', function (done) {
+				transaction.fee = -1.2;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Expected type integer but found type number');
+					badTransactions.push(transaction);
+					done();
+				});
+			});
+
+			it('using more than maximum should fail', function (done) {
+				transaction.fee = Number(new BigNumber(constants.totalAmount)+ 1);
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.have.string('Invalid transaction body - Failed to validate transaction schema: Value '+transaction.fee+' is greater than maximum 10000000000000000');
+					badTransactions.push(transaction);
+					done();
+				});
+			});
+		});
+
+		describe('sending transactions with wrong recipientId values', function () {
+
+			it('using null should fail', function (done) {
+				transaction.recipientId = null;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction id');
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('without property should fail', function (done) {
+				delete transaction.recipientId;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction id');
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using undefined should fail', function (done) {
+				transaction.recipientId = undefined;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction id');
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using NaN should fail', function (done) {
```

Comment:
Don't like. Let's try to improve more with https://mochajs.org/#dynamically-generating-tests